airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/utils/connector_runner.py

```diff
@@ -109,7 +112,15 @@ def run(self, cmd, config=None, state=None, catalog=None, raise_container_error:
             for line in self.read(container, command=cmd, with_ext=raise_container_error):
                 f.write(line.encode())
                 try:
-                    yield AirbyteMessage.parse_raw(line)
+                    airbyte_message = AirbyteMessage.parse_raw(line)
+                    if (
+                        airbyte_message.type is AirbyteMessageType.CONTROL
+                        and airbyte_message.control.type is OrchestratorType.CONNECTOR_CONFIG
+                    ):
+                        self._persist_new_configuration(
+                            airbyte_message.control.connectorConfig.config, int(airbyte_message.control.emitted_at)
+                        )
+                    yield airbyte_message
```

Comment:
why do we want to yield if it's a control message?

Reply:
The original logic yield all outputted messages, whatever their type. I did not want to change this to avoid unpredicted side effects...